src/common/utils/IndexKeyUtils.h

```diff
@@ -203,9 +223,11 @@ class IndexKeyUtils final {
     auto val = *reinterpret_cast<const double*>(v);
     val = folly::Endian::big(val);
     if (val < 0) {
-      auto i = *reinterpret_cast<const int64_t*>(&val);
+      auto* c1 = reinterpret_cast<const char*>(&val);
+      auto i = *reinterpret_cast<const int64_t*>(c1);
       i = -(std::numeric_limits<int64_t>::max() + i);
-      val = *reinterpret_cast<const double*>(&i);
+      auto* c2 = reinterpret_cast<const char*>(&i);
+      val = *reinterpret_cast<const double*>(c2);
```

Comment:
Explain it.

Reply:
See https://stackoverflow.com/questions/4163126/dereferencing-type-punned-pointer-will-break-strict-aliasing-rules-warning.

Reply:
I find nebula has configured the option:
```
option(ENABLE_STRICT_ALIASING           "Build with -fstrict-aliasing" OFF)
```
```
if(NOT ENABLE_STRICT_ALIASING)
    add_compile_options(-fno-strict-aliasing)
else()
    add_compile_options(-fstrict-aliasing)
endif()
```
I don't know why this option doesn't work now